### PR TITLE
Add three extensions for JsonToKotlinClass plugin

### PR DESCRIPTION
**Pull request by varenyzc**
CamelCaseSupport:  It can type camel-case property.
BuildFromJsonSupport :  It can type a companion object with a function what can build from json.
NeedNonNullableClassesSupport: It can make the classes that you type in the extensions input area non nullabe.

### CODE REVIEW

**Comment by wuseal** [comment]
@varenyzc Greate! Amazing me, you do these so fast 😱

Also we need unit test for each new feature.

Please make up the test cases. Thank you 😁

**Comment by varenyzc** [comment]
@wuseal okay，i will make it tomorrow😁

**Comment by varenyzc** [comment]
@wuseal I have completed the test case. Should I make a new pull request?

**Comment by wuseal** [comment]
@varenyzc  No need, You could just push your test cases to branch varenyzc:master is OK

File: src/main/kotlin/extensions/yuan/varenyzc/BuildFromJsonObjectSupport.kt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import javax.swing.JPanel
+
+object BuildFromJsonObjectSupport : Extension() {
+
+    const val configKey = "top.varenyzc.build_from_json_enable"
+
+    override fun createUI(): JPanel {
+        return jHorizontalLinearLayout {
+            jCheckBox(
+                "Enable build from JsonObject",
```
Comment (wuseal) [inline src/main/kotlin/extensions/yuan/varenyzc/BuildFromJsonObjectSupport.kt:18]: How about named it as ```add `fromJson` Function ```? It's not clear what is `Enable build from JsonObject`
Also add a small doc for this option can make people understand this feature fast as I think, you could reference https://github.com/wuseal/JsonToKotlinClass/blob/aabaa6a86bb5bb6d83b737904d285331c277ad91/src/main/kotlin/extensions/jose/han/ParcelableAnnotationSupport.kt#L27 to adding a Link

File: src/main/kotlin/extensions/yuan/varenyzc/CamelCaseSupport.kt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    /**
+     * Config key can't be private, as it will be accessed from `library` module
+     */
+    @Suppress("MemberVisibilityCanBePrivate")
+    const val configKey = "top.varenyzc.camel_case_enable"
+
+    override fun createUI(): JPanel {
+        return jHorizontalLinearLayout {
+            jCheckBox(
+                "Enable Camel-Case",
```
Comment (wuseal) [inline src/main/kotlin/extensions/yuan/varenyzc/CamelCaseSupport.kt:23]: I prefer named it as `Let properties' name to be camel case

File: src/main/kotlin/extensions/yuan/varenyzc/NeedNonNullableClassesSupport.kt [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            addFocusLostListener {
+                if (getConfig(prefixKeyEnable).toBoolean()) {
+                    setConfig(prefixKey, text)
+                }
+            }
+            maximumSize = JBDimension(400, 30)
+        }
+
+        return jHorizontalLinearLayout {
+            jCheckBox("Need NonNullable classes: ", getConfig(prefixKeyEnable).toBoolean(), { isSelected ->
```
Comment (wuseal) [inline src/main/kotlin/extensions/yuan/varenyzc/NeedNonNullableClassesSupport.kt:27]: For this name, I really don't know what it does until I see your test case😂, Try to think of a better name and paste a link to the UI to show the usage and demo

Comment (varenyzc) [reply src/main/kotlin/extensions/yuan/varenyzc/NeedNonNullableClassesSupport.kt:27]: 😂,okay, i will think about it.Maybe my English is so poor.

**Comment by wuseal** [comment]
@varenyzc  Well done! Thanks very much that you provided these awesome features for JTK👏👏👏👏👍

**Review by wuseal — APPROVED** [review]
LGTM

**Comment by varenyzc** [comment]
@wuseal This is my first time to make a PR for open source project.Thanks for your review！

**Comment by wuseal** [comment]
Published in 3.7.1-EAP-3

**Pull request merged**